### PR TITLE
feat(plugins): add event logs

### PR DESCRIPTION
**Pull request by flegastelois**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | N/A

It's time to logs all important event on plugins! (install/uninstall/activate/disable)

### CODE REVIEW

**Review by trasher — APPROVED** [review]

File: src/Plugin.php [hunk 1]
```diff
@@ -798,6 +799,14 @@ public function uninstall($ID)
 
             $type = INFO;
             $message = sprintf(__('Plugin %1$s has been uninstalled!'), $this->fields['name']);
+
+            Event::log(
+                '',
+                Plugin::class,
+                4,
```
Comment (cedric-anne) [inline src/Plugin.php:806]: Level should probably be 3.

Maybe you can introduce some constants to make log levels values more readable.

```suggestion
                3,
```

File: src/Plugin.php [hunk 2]
```diff
@@ -845,26 +854,34 @@ public function install($ID, array $params = [])
...
                         ]);
-                        $message = sprintf(__('Plugin %1$s has been installed and must be configured!'), $this->fields['name']);
+                        $log_message = sprintf(__('Plugin %1$s has been installed and must be configured!'), $this->fields['name']);
+                        $message = $log_message;
                     }
                     self::doHook(Hooks::POST_PLUGIN_UNINSTALL, $this->fields['directory']);
+                    Event::log(
+                        '',
+                        Plugin::class,
+                        4,
```
Comment (cedric-anne) [inline src/Plugin.php:881]:
```suggestion
                        3,
```

File: src/Plugin.php [hunk 3]
```diff
@@ -965,12 +982,22 @@ public function activate($ID)
...
-                    sprintf(__('Plugin %1$s has been activated!'), $this->fields['name']),
+                    $log_message,
                     true,
                     INFO
                 );
 
+                Event::log(
+                    '',
+                    Plugin::class,
+                    4,
```
Comment (cedric-anne) [inline src/Plugin.php:996]:
```suggestion
                    3,
```

File: src/Plugin.php [hunk 4]
```diff
@@ -1021,12 +1048,22 @@ public function unactivate($ID)
...
-                sprintf(__('Plugin %1$s has been deactivated!'), $this->fields['name']),
+                $log_message,
                 true,
                 INFO
             );
 
+            Event::log(
+                '',
+                Plugin::class,
+                4,
```
Comment (cedric-anne) [inline src/Plugin.php:1062]:
```suggestion
                3,
```

File: src/Plugin.php [hunk 5]
```diff
@@ -1070,6 +1107,14 @@ public function unactivateAll()
         if (isset($_SESSION['glpimenu'])) {
             unset($_SESSION['glpimenu']);
         }
+
+        Event::log(
+            '',
+            Plugin::class,
+            4,
```
Comment (cedric-anne) [inline src/Plugin.php:1114]:
```suggestion
            3,
```

File: src/Plugin.php [hunk 6]
```diff
@@ -1083,8 +1128,17 @@ public function clean($ID)
...
         if ($this->getFromDB($ID)) {
             $this->unload($this->fields['directory']);
+            $log_message = sprintf(__('Plugin %1$s cleaned!'), $this->fields['directory']);
             self::doHook(Hooks::POST_PLUGIN_CLEAN, $this->fields['directory']);
             $this->delete(['id' => $ID]);
+
+            Event::log(
+                '',
+                Plugin::class,
+                4,
```
Comment (cedric-anne) [inline src/Plugin.php:1138]:
```suggestion
                3,
```

File: src/Plugin.php [hunk 7]
```diff
@@ -1070,6 +1107,14 @@ public function unactivateAll()
         if (isset($_SESSION['glpimenu'])) {
             unset($_SESSION['glpimenu']);
         }
+
+        Event::log(
+            '',
+            Plugin::class,
+            4,
+            "setup",
+            __('All plugins have been disabled by the GLPI update process.')
```
Comment (cedric-anne) [inline src/Plugin.php:1116]: Anyone could call this method. We do not want people to blame update process if a custom script calls this method.
```suggestion
            __('All plugins have been disabled.')
```

**Review by cedric-anne — CHANGES_REQUESTED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Review by trasher — APPROVED** [review]

**Pull request merged**